diff -upr cron-3.0pl1.orig/pathnames.h cron-3.0pl1/pathnames.h
--- cron-3.0pl1.orig/pathnames.h	2008-05-21 10:42:06.064487361 -0500
+++ cron-3.0pl1/pathnames.h	2008-05-21 10:41:34.412450000 -0500
@@ -75,7 +75,7 @@
 			 * environment variable specified.
 			 */
 #if defined(DEBIAN)
-# define EDITOR "/usr/bin/editor"
+# define EDITOR "/usr/bin/sensible-editor"
 #elif defined(_PATH_VI)
 # define EDITOR _PATH_VI
 #else
